### PR TITLE
(chore) disallow unknown fields29 fee ack

### PR DESCRIPTION
**Pull request by bznein**
## Description

closes: #6921

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/29-fee/types/encoding.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package types
```
Comment (bznein) [inline modules/apps/29-fee/types/encoding.go:1]: This is a copy of a file that already exists. We can definitely merge them together though I'm not sure where/how. Happy to do it now/in another PR if people feel strongly

File: modules/apps/29-fee/ibc_middleware.go [hunk 2]
```diff
@@ -250,7 +250,7 @@ func (im IBCMiddleware) OnAcknowledgementPacket(
 	}
 
 	var ack types.IncentivizedAcknowledgement
-	if err := types.ModuleCdc.UnmarshalJSON(acknowledgement, &ack); err != nil {
+	if err := ack.UnmarshalJSON(acknowledgement); err != nil {
```
[the remark this answers is not in the document]
Comment (bznein) [reply modules/apps/29-fee/ibc_middleware.go:253]: ah! of course! How did I miss that? :D

**Review by damiannolan — APPROVED** [review]
type_safety 🏗️

File: modules/apps/29-fee/ibc_middleware_test.go [hunk 3]
```diff
@@ -1592,3 +1592,34 @@ func (suite *FeeTestSuite) TestPacketDataUnmarshalerInterfaceError() {
...
+		{
+			"failure: unknown fields",
+			[]byte(`{"app_acknowledgement": "eyJyZXN1bHQiOiJiVzlqYXlCaFkydHViM2RzWldsblpXMWxiblE9In0=", "forward_relayer_address": "relayer", "underlying_app_success": true, "extra_field": "foo"}`),
+			false,
+		},
+	}
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			ack := types.IncentivizedAcknowledgement{}
+			err := ack.UnmarshalJSON(tc.ackBytes)
```
[the remark this answers is not in the document]
Comment (bznein) [reply modules/apps/29-fee/ibc_middleware_test.go]: Yep good idea! done

**Pull request merged**